### PR TITLE
COMP: GDCM: fix CMake Deprecation Warning

### PR DESCRIPTION
**Pull request by blowekamp**
Use cmake_minimum_required to set min and policy_max

    CMake version >3.30 produce a deprecated awarning when only, a min
    argument less that 3.10 is pass to cmake_minimum_required. Using the
    additional policy_max version range specified the max policy setting
    at the same time.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
upstream pull request: https://github.com/malaterre/GDCM/pull/181

File: Modules/ThirdParty/GDCM/src/gdcm/CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,7 @@
-cmake_minimum_required(VERSION 3.9.2 FATAL_ERROR)
+set(GDCM_MAX_VALIDATED_CMAKE_VERSION "3.13.4")
```
Comment (dzenanz) [inline Modules/ThirdParty/GDCM/src/gdcm/CMakeLists.txt:1]: Is this the only file we bundle in ITK? Upstream PR has more changes.

**Review by dzenanz — APPROVED** [review]

File: Modules/ThirdParty/GDCM/src/gdcm/CMakeLists.txt (hunk 1 above)
Comment (blowekamp) [reply Modules/ThirdParty/GDCM/src/gdcm/CMakeLists.txt:1]: In our bundled versions, the best I can tell that code is not reachable.

**Comment by hjmjohnson** [comment]
@blowekamp Is this ready to be taken out of draft mode?

**Comment by blowekamp** [comment]
The upstream PR has not been reviewed/merged yet. I ideally our local changes will match what is accepted upstream to avoid conflicts.

**Comment by blowekamp** [comment]
@malaterre Can you please review?

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**